### PR TITLE
Add search in Attributes panel 

### PR DESCRIPTION
**Pull request by MrRamka**
Implementation https://github.com/microsoft/charticulator/issues/834
search

image

### CODE REVIEW

File: src/app/views/panels/widgets/controls/collapsiblePanel.tsx [hunk 1]
```diff
@@ -9,28 +9,61 @@ import {
...
+  store,
 }) => {
-  const [groupState, setIsCollapsed] = React.useState<boolean>(
-    isCollapsed === undefined ? false : isCollapsed
+  const collapsePanel = store
+    ? false
+    : isCollapsed === undefined
+    ? false
+    : isCollapsed;
+  const [groupState, setIsCollapsed] = React.useState<boolean>(collapsePanel);
```
Comment (jrthompson33) [inline src/app/views/panels/widgets/controls/collapsiblePanel.tsx:45]: Consider changing the set state function to be `setGroupState` for clarity.

File: src/core/prototypes/legends/categorical_legend.ts [hunk 2]
```diff
@@ -303,6 +303,7 @@ export class CategoricalLegendClass extends LegendClass {
               ],
               options: [OrientationType.VERTICAL, OrientationType.HORIZONTAL],
               label: strings.objects.legend.orientation,
+              searchSection: strings.objects.legend.categoricalLegend,
```
Comment (jrthompson33) [inline src/core/prototypes/legends/categorical_legend.ts:306]: Should categoricalLegend have a search header?

File: src/core/prototypes/legends/color_legend.ts [hunk 3]
```diff
@@ -136,6 +136,7 @@ export class NumericalColorLegendClass extends LegendClass {
                 : strings.objects.height,
               updownTick: 10,
               showUpdown: true,
+              searchSection: strings.objects.legend.numericalColorLegend,
```
Comment (jrthompson33) [inline src/core/prototypes/legends/color_legend.ts:139]: Should numericalColorLegend have a search header?

File: src/core/prototypes/legends/custom_legend.ts [hunk 4]
```diff
@@ -67,6 +67,7 @@ export class CustomLegendClass extends CategoricalLegendClass {
         manager.vertical(
           manager.label(strings.objects.colors, {
             addMargins: true,
+            searchSection: strings.objects.colors,
```
Comment (jrthompson33) [inline src/core/prototypes/legends/custom_legend.ts:70]: Same here, should this be a verticalGroup with a search header

File: src/core/prototypes/marks/nested_chart.ts [hunk 5]
```diff
@@ -81,37 +82,44 @@ export class NestedChartElementClass extends EmphasizableMarkClass<
...
+            hints: { autoRange: true, startWithZero: "always" },
+            acceptKinds: [Specification.DataKind.Numerical],
+            defaultAuto: true,
+            searchSection: strings.objects.nestedChart.sizeAnsShape,
+          }),
+          manager.mappingEditor(strings.objects.height, "height", {
+            hints: { autoRange: true, startWithZero: "always" },
+            acceptKinds: [Specification.DataKind.Numerical],
+            defaultAuto: true,
+            searchSection: strings.objects.nestedChart.sizeAnsShape,
```
Comment (jrthompson33) [inline src/core/prototypes/marks/nested_chart.ts:99]: Type-o? Should this be `sizeAnsShape`

File: src/strings.ts [hunk 6]
```diff
@@ -591,10 +596,17 @@ export const strings = {
       second: "Second",
       menuSuffix: " Derived columns ",
     },
+    nestedChart: {
+      sizeAnsShape: "Size & Shape",
```
Comment (jrthompson33) [inline src/strings.ts:600]: Type-o looks like this should be `sizeAndShape`

**Review by jrthompson33 — COMMENTED** [review]
Looks good! Please see minor comments and one type-o to change

**Comment by MrRamka** [comment]
Thank you! All comments have been fixed

**Review by bongshin — APPROVED** [review]

**Pull request merged**